docs/en/operations/settings/settings.md

```diff
@@ -5418,6 +5418,12 @@ When set to `false` than all attempts are made with identical timeouts.
 
 Default value: `true`.
 
+## uniform_snowflake_conversion_functions {#uniform_snowflake_conversion_functions}
+
+If set to `true`, then functions `snowflakeIDToDateTime`, `snowflakeIDToDateTime64`, `dateTimeToSnowflakeID`, and `dateTime64ToSnowflakeID` are enabled, and functions `snowflakeToDateTime`, `snowflakeToDateTime64`, `dateTimeToSnowflake`, and `dateTime64ToSnowflake` are disabled (and vice versa if set to `false`).
```

Comment:
Doesn't it make more sense to have `snowflakeIDToDateTime`, `snowflakeIDToDateTime64`, `dateTimeToSnowflakeID`, and `dateTime64ToSnowflakeID` always enabled and have a setting such as `allow_deprecated_snowflake_conversion_functions`?

Otherwise you need to migrate everything at once or change the setting per query, which is not ideal

Reply:
I was actually not sure. There is a fine balance between
- giving more time for migration like you propose (which means migration may never happen) and
- "encouraging" the migration like in the current implementation (which creates an inconvenience for users).

But I guess what you propose makes sense, will change that (tomorrow likely).

Reply:
https://github.com/ClickHouse/ClickHouse/pull/65522